src/pi-src/data_collection.cpp

```diff
@@ -14,13 +15,33 @@ Created: 2018-02-10
 #include <stdio.h>
 #include <string.h>
 #include "data_collection.h"
+#include "uid.h"
 #include "spi.h"
 #include "util.h"
 
-#define NUM_PINS 5 /* TODO: Define this elsewhere, globally */
-
+#define NUM_PINS 4 /* TODO: Define this elsewhere, globally */
+#define PING_BUF_LEN 16 /* TODO: Define as config param */
 int DISPLAY_PINGS = 0;
 
+/**
+ * As pings come in, we need to store them. We need at least 2 pings per pin
+ * to run the direction algorithm, but we also need to account for the fact
+ * that we may get some noise between two pings. Thus we have a small buffer
+ * of size PING_BUF_LEN for each pin. As pings come in, they are stored in the
+ * appropriate buffer. Later, we search through all these buffers to find
+ * matching pings that we can run triangulation on.
+ * We insert at the "end" of this circular buffer every time and erase pings
+ * once they have been used. If a ping becomes "stale," meaning that it has
+ * been in the buffer for a while without being used in a calculation, it will
+ * eventually be overwritten with fresh data as the insert head of the buffer
```

Comment:
How long is awhile? And how is that being kept track of?

Reply:
It depends on how quickly data comes in. It just gets overwritten when the buffer wraps back around, so if data is coming in quickly that could happen fairly often. However, if we don't hear anything for a while, it'll just sit there.

If we get 1 ping per second, then it will take `PING_BUF_LEN` seconds before the buffer loops back around to where it started. Does that make sense?